FE-1207 - Remove the feature flag and remove the sending domain and tracking domain pages

shrirupa opened the pull request:
FE-1207 - Remove the feature flag and remove the sending domain and tracking domain pages from nav in hibana

### What Changed
 -  Removed the feature flag - allow_domains_v2
 - Removed Sending and Tracking Domain from navigation in Hibana
 - Removed the routes to sending domain and tracking domain in hibana
 - OG theme still has the old sending and tracking domain pages intact

### How To Test
 - Check on a account with allow_domain_v2 not enabled, domains page is visible in the nav
 - Check in OG version the sending and tracking domain pages are visible

### To Do
- [ ] Address feedback

nicklemmon commented on cypress/fixtures/account/200.get.has-domains-v2.json:
```diff
@@ -5,7 +5,6 @@
-        "allow_domains_v2": true
       }
     }
   }
```
I think this whole fixture can be removed? (and any references to it)

nicklemmon reviewed (changes requested):
@shrirupa Just had one tiny thing - we can remove that fixture from the repo entirely (and any references to it). Everything else looks good.

shrirupa commented:
@nicklemmon Thanks for the feedback! This is ready for another pass.

I have added a few more things to the PR - 
* Replaced the Select in Filters with ListBox (UX wanted it to be a ListBox instead of Select)
* Added a component wrapper for ListBox
* Fixed the spacing bug that you found.

nicklemmon commented on src/pages/domains/components/TableFilters.js:
```diff
@@ -95,13 +95,17 @@ function SortSelect({ options, onChange, disabled }) {
   return (
     <StyledGridCell>
-      <Select
+      <ListBox
```
The keyboard controls for this aren't working? They work here (focus with your keyboard, hit the "down" arrow and you can select an option)
https://design.sparkpost.com/components/listbox

Not sure if their example is set up with Downshift.

nicklemmon commented on cypress/tests/integration/domains/listPage.spec.js:
```diff
@@ -288,7 +288,8 @@ describe('The domains list page', () => {
 
-        cy.findByLabelText('Sort By').select('Domain Name (A - Z)');
+        cy.findByLabelText('Sort By').click();
+        cy.findAllByText('Domain Name (A - Z)').click();
```
You should be able to use `cy.findByRole('option',  { name: 'Domain Name (A - Z)' }).click()`

nicklemmon commented on src/components/matchbox/ListBox.js:
```diff
@@ -0,0 +1,35 @@
+import { ListBox as HibanaListBox } from '@sparkpost/matchbox-hibana';
+import { useHibana } from 'src/context/HibanaContext';
+
+function ListBox(props) {
+  const [state] = useHibana();
+  const { isHibanaEnabled } = state;
+
+  if (!isHibanaEnabled) {
+    throw new Error(
+      'ListBox component not available in original matchbox. Please remove or restrict to Hibana',
+    );
+  }
+
+  return <HibanaListBox {...props} />;
+}
+
+function Option(props) {
```
I think we need to forward refs for this wrapper:
<img width="625" alt="Screen Shot 2020-11-10 at 2 15 22 PM" src="https://user-images.githubusercontent.com/3613392/98722577-3ac84200-235f-11eb-8ed5-1f18b1221596.png">

shrirupa replied:
Thanks for catching this!

nicklemmon reviewed (changes requested):
@shrirupa A couple of things - mainly related to the `ListBox`. I think the keyboard controls might not be working because of the lack of ref forwarding?

There are also a few errors in the console I'm seeing you should take a look at:
<img width="557" alt="Screen Shot 2020-11-10 at 2 18 07 PM" src="https://user-images.githubusercontent.com/3613392/98722872-9b577f00-235f-11eb-9333-22dec2ef1c43.png">
<img width="557" alt="Screen Shot 2020-11-10 at 2 18 07 PM" src="https://user-images.githubusercontent.com/3613392/98722891-a1e5f680-235f-11eb-91e9-0d06f3c6fb50.png">
<img width="640" alt="Screen Shot 2020-11-10 at 2 18 45 PM" src="https://user-images.githubusercontent.com/3613392/98722919-ab6f5e80-235f-11eb-958b-45b29250d99b.png">
<img width="539" alt="Screen Shot 2020-11-10 at 2 19 06 PM" src="https://user-images.githubusercontent.com/3613392/98722942-b4603000-235f-11eb-8355-83bfaacab0e0.png">

shrirupa commented:
@nicklemmon Thanks again for catching that forwardRef issue. This is ready for another pass.

nicklemmon reviewed (approved):
@shrirupa Looks good! The keyboard controls are working now for the `ListBox`. Nice job.

Let me know when you want to deploy.
